cmake/onnxruntime_snpe_provider.cmake

```diff
@@ -0,0 +1,50 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+set(PROVIDERS_SNPE onnxruntime_providers_snpe)
+add_compile_definitions(USE_SNPE=1)
+
+file(GLOB_RECURSE
+  onnxruntime_providers_snpe_cc_srcs CONFIGURE_DEPENDS
+  "${ONNXRUNTIME_ROOT}/core/providers/snpe/*.h"
+  "${ONNXRUNTIME_ROOT}/core/providers/snpe/*.cc"
+)
+
+if(ANDROID)
+  # Specify the link libraries
+  set(SNPE_NN_LIBS ${SNPE} libc++_shared.so)
+else()
+  set(SNPE_NN_LIBS ${SNPE})
+endif()
+
+file(TO_CMAKE_PATH ${SNPE_ROOT} SNPE_ROOT)
+get_filename_component(SNPE_CMAKE_DIR ${SNPE_ROOT} ABSOLUTE)
+file(TO_CMAKE_PATH "${SNPE_CMAKE_DIR}/lib/${SNPE_ARCH_ABI}" SNPE_LIB_DIR)
+file(TO_NATIVE_PATH ${SNPE_LIB_DIR} SNPE_NATIVE_DIR)
+message(STATUS "Looking for SNPE library in ${SNPE_NATIVE_DIR}")
+find_library(SNPE NAMES snpe SNPE libSNPE.so PATHS "${SNPE_NATIVE_DIR}" "${SNPE_ROOT}" NO_DEFAULT_PATH NO_CMAKE_FIND_ROOT_PATH REQUIRED)
+file(GLOB SNPE_SO_FILES LIST_DIRECTORIES false "${SNPE_CMAKE_DIR}/lib/${SNPE_ARCH_ABI}/*.so" "${SNPE_CMAKE_DIR}/lib/${SNPE_ARCH_ABI}/*.dll")
+# add dsp skel files to distribution
+file(GLOB SNPE_DSP_FILES LIST_DIRECTORIES false "${SNPE_CMAKE_DIR}/lib/dsp/*.so")
```

Comment:
What do you mean "add dsp skel files to distribution"? Do do need to include their file in our package?

Reply:
just copy them to build out folder, we won't include any dependency lib in our package